### ISSUE TITLE
Updating reqwest to 0.11.22 causes macos build failure, requires SystemConfiguration

### ISSUE THREAD

**Issue by jsantell** [subconsciousnetwork/noosphere#686]
In some dev work, I updated `reqwest` from 0.11.20 to 0.11.22 and ran into [failures](https://github.com/subconsciousnetwork/noosphere/actions/runs/6566865808/job/17840254182?pr=549) in a x86_64 macos build:

```
Undefined symbols for architecture x86_64:
  "_SCDynamicStoreCopyProxies", referenced from:
      __ZN20system_configuration13dynamic_store14SCDynamicStore11get_proxies17ha9cee4367b0c1fcbE in libnoosphere.a(system_configuration-46c6672ca6db6ca9.system_configuration.b48ec42a4ca5d16c-cgu.0.rcgu.o)
  "_SCDynamicStoreCreateWithOptions", referenced from:
      __ZN20system_configuration13dynamic_store14SCDynamicStore6create17he63aeeeb3840fdc4E in libnoosphere.a(system_configuration-46c6672ca6db6ca9.system_configuration.b48ec42a4ca5d16c-cgu.0.rcgu.o)
  "_SCNetworkInterfaceCopyAll", referenced from:
      __ZN20system_configuration21network_configuration14get_interfaces17ha41c0e3fd9b2cb6eE in libnoosphere.a(system_configuration-46c6672ca6db6ca9.system_configuration.b48ec42a4ca5d16c-cgu.0.rcgu.o)
  "_SCNetworkInterfaceGetBSDName", referenced from:
      __ZN20system_configuration21network_configuration18SCNetworkInterface8bsd_name17h45a02c56c7f559b2E in libnoosphere.a(system_configuration-46c6672ca6db6ca9.system_configuration.b48ec42a4ca5d16c-cgu.0.rcgu.o)
  "_SCNetworkInterfaceGetInterfaceType", referenced from:
      __ZN20system_configuration21network_configuration18SCNetworkInterface21interface_type_string17hd31e474b334507ecE in libnoosphere.a(system_configuration-46c6672ca6db6ca9.system_configuration.b48ec42a4ca5d16c-cgu.0.rcgu.o)
  "_SCNetworkInterfaceGetLocalizedDisplayName", referenced from:
      __ZN20system_configuration21network_configuration18SCNetworkInterface12display_name17h94d4d8074b27dd32E in libnoosphere.a(system_configuration-46c6672ca6db6ca9.system_configuration.b48ec42a4ca5d16c-cgu.0.rcgu.o)
  "_SCNetworkReachabilityCreateWithAddress", referenced from:
```

In https://github.com/seanmonstar/reqwest/pull/1955, [system-configuration](https://github.com/mullvad/system-configuration-rs) was added to reqwest, requiring linking to the SystemConfiguration framework. The transitive dependency seems to link the framework in its [build script](https://github.com/mullvad/system-configuration-rs/blob/main/system-configuration-sys/build.rs) at a glance, but we hit undefined symbol errors. Need to evaluate if this is an intended required change, and if so, learn more about if we want to require [SystemConfiguration.framework](https://developer.apple.com/documentation/systemconfiguration) on macos.

---

Upstream bug in system-configuration: https://github.com/mullvad/system-configuration-rs/issues/41
Patch for iOS, to investigate: https://github.com/mullvad/system-configuration-rs/pull/42/files

---

Filed upstream Reqwest bug: https://github.com/seanmonstar/reqwest/issues/2006

**Comment by jsantell** [comment]
Fixed upstream, should be fixed in next reqwest release (0.11.25)

**Comment by jsantell** [comment]
The macos system configuration dependency was made optional in 0.12 https://github.com/seanmonstar/reqwest/pull/2185
